core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -30,9 +30,10 @@
 public class HasParentQueryParser implements QueryParser<HasParentQueryBuilder>  {
 
     private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
-    private static final ParseField SCORE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
-    private static final ParseField TYPE_FIELD = new ParseField("parent_type", "type");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
```

Comment:
Can you switch this around and use the preferred name as first constructor argument? This way it looks like there's something special with this field, which I guess its not.